### ISSUE TITLE
Need a "web" installer for desktop app bundling

### ISSUE THREAD

**Issue by rickbrew** [dotnet/core-setup#4854]
I'm in the early stages of researching and planning for migrating Paint.NET (https://www.getpaint.net/) from .NET Framework 4.7 to .NET Core 3. I noticed that Core does not have a "web installer" -- that is, a small stub installer (~1.4MB for .NET Framework 4.7.2) that downloads the rest of what's needed on-demand from Microsoft's servers.

This is important for Paint.NET for several reasons. I need to be able to continue providing a "small" download package (currently 7.5MB) that won't greatly increase the bandwidth demands on my servers, and that will skip the .NET Core download when it's already installed on the system. This is also a big win for my users, many of whom do not yet have gigabit Internet ;)

Without this solution, the Paint.NET package would balloon in size by at least 30MB! And this doesn't yet take into account the additional size that WinForms and WPF will add to the package.

IIRC, the old releases of the .NET Framework "Client Profile" installer (3.5 SP1, right?) had instructions for how to modify the installer in order to download additional dependencies like the Visual C++ runtimes. I realize time and resources are always in short supply, so even instructions on how to modify the existing .NET Framework installer to instead pull down .NET Core would be immensely helpful. (other solutions and advice are welcome, too)

I'm not too worried about this for the Microsoft Store release of Paint.NET, but it's critical for the "Classic" desktop version.

**Comment by rickbrew** [comment]
btw, I do expect the Paint.NET download size to increase during the migration from Framework to Core. Going from 7.5MB to even ~15MB would be alright, but 40+ would be a problem.

**Comment by rakeshsinghranchi** [comment]
cc @vivmishra  @leecow @dleeapho

**Comment by dleeapho** [comment]
@rickbrew thanks for surfacing a concrete scenario for a web installer for .NET Core deployment on the desktop. I'm assuming you are considering a framework-dependent deployment for Paint.NET, but have you considered a self-contained deployment? In particular the trimming you are looking for might be achieved through [IL Linker](https://github.com/dotnet/core/issues/915).

**Comment by rickbrew** [comment]
@dleeapho Yes, FDD is what I'm going for. However, Paint.NET supports plugins and they will need access to portions of the framework that I can't really predict, so I can't just link-and-trim.
